models/productDao.js

```diff
@@ -0,0 +1,61 @@
+const { teaDataSource } = require("./datasource.js");
+
+const getProduct = async (start, pageSize, category, sub, type, sort) => {
+  let whereCondition = category ? `WHERE` : type ? `WHERE` : "";
+  let andCondition = category && type ? `AND` : sub && type ? `AND` : "";
+  const categoryCondition = category ? `c.name='${category}'` : "";
+  const subCondition = sub ? `AND s.name='${sub}'` : "";
+  const typeCondition = type[0]
+    ? type[1]
+      ? type[2]
+        ? type[3]
+          ? `(t.type='${type[0]}' OR t.type='${type[1]}' OR t.type='${type[2]}',t.type='${type[3]}')`
+          : `(t.type='${type[0]}' OR t.type='${type[1]}' OR t.type='${type[2]}')`
+        : `(t.type='${type[0]}' OR t.type='${type[1]}')`
+      : `(t.type='${type[0]}')`
+    : "";
+  let sortCondition = "";
+  if (sort === "price_asc") {
+    sortCondition = `ORDER BY p.discount_price ASC`;
+  } else if (sort === "price_desc") {
+    sortCondition = `ORDER BY p.discount_price DESC`;
+  } else if (sort === "new_arrival") {
+    sortCondition = `ORDER BY p.created_at DESC`;
+  }
```

Comment:
객체를 통해 매핑하는 방법은 다음과 같이 적을 수 있습니다:
`sortSet`이라는 객체의 key 값으로 클라이언트가 요청하는 sort value를 적어줍니다.
만약 클라이언트가 요청하는 sort value가 sortSet의 key값으로 있다면 ( == `sortSet[sort]`) 해당 쿼리문이 변수에 할당되도록 구현할 수 있습니다.


```suggestion
  const sortSet = {
    new_arrival: "ORDER BY p.created_at DESC",
    price_desc: "ORDER BY p.discount_price DESC",
    price_asc: "ORDER BY p.discount_price ASC",
  };
  
  const sortCondition = sortSet[sort];
```